### PR TITLE
&& Nicoleta B

### PR DESCRIPTION
**Pull request by nbrandolini**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, how did you try querying the API? |I looked at Edamam API documentation and used Postman to query for recipes and check the JSON for formatting  my Recipe.new |
Describe your API Wrapper.  How did you decide on the methods you created? | I created two methods one to search for recipes and the other to find a recipe and I use two different API requests. The search method uses a q search and the find method uses an r search in the Postman app.|
Describe an edge case or failure case test you wrote for your API Wrapper. | An edge case I wrote for the API Wrapper is to return nil for bogus arguments|
Explain how VCR aids in testing an API. | VCR helps because it reduces the dependencies on external sources. Also, it will not count against the # of free calls to the API.|
What is the Heroku URL of your deployed application? |https://ada-muncher.herokuapp.com/ |
Provide a link to the Trello board you used |https://trello.com/b/oNIsKo9C/api-muncher |

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class RecipesController < ApplicationController
+
+  def home
+  end
+
+  def index
+    @recipes = ApiMuncherWrapper.search_recipes(params[:search]).paginate(:page => params[:page],:per_page => 10)
```
Comment (CheezItMan) [inline app/controllers/recipes_controller.rb:7]: You need to have some sort of check to verify that you get an array back here.

File: test/lib/api_muncher_wrapper_test.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        recipes.each do |recipe|
+          recipe.must_be_instance_of Recipe
+        end
+      end
+    end
+
+    it "will return an empty array if recipe does not exist" do
+
+      VCR.use_cassette("recipes") do
+        recipes = ApiMuncherWrapper.search_recipes("abbbcd1234567")
```
Comment (CheezItMan) [inline test/lib/api_muncher_wrapper_test.rb:23]: Your API returns nil instead of an empty array

File: test/controllers/recipes_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    end
+  end
+
+
+  it "should get index" do
+    VCR.use_cassette("index_action") do
+      get recipes_path, params: { search: "pesto" }
+      must_respond_with :success
+
+      get recipes_path, params: { search: " " }
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:18]: No error handling for if the search returns no results?

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** | 
Git hygiene | Good number of commits, some funny commit messages, "OMG Show worksgit add ."
Comprehension questions | Check
**General** | 
Rails fundamentals (RESTful routing, use of named paths) | Check
Semantic HTML | Why embed the `footer` element the `main`?  Mostly just curious, otherwise well done
Errors are reported to the user | Check, both for the search results and the show page
API Wrapper to handle the API requests | Check
Controller testing | Check, small notes here for checking for negative cases and handling them more throughly
Lib testing | Small broken test with getting a recipe that doesn't exist.  
Search Functionality | Check
List Functionality |  Check
Show individual item functionality (link to original recipe opens in new tab) | Check
**Styling** | 
Responsive layout | The search results page is responsive, the show page is not.
List View shows 10 items at a time/pagination | Check
The app is styled to create an attractive user interface | The show page could definitely use some work, notably the list items still have bullets!
**API Features** | 
The App attributes Edaman | Check although it lacks the image
The VCR casettes do not contain the API key | check
**External Resources** | 
Link to Trello Board | **I Do not have access to the board**
Link to deployed app on Heroku | Check
**Overall** | Overall well done, you've got a little work to do on testing and handling error-cases, but you hit all the major learning goals.
